### PR TITLE
adding e2e matrix tests for forkliftci

### PR DESCRIPTION
**Pull request by eslutsky**
adding e2e forklift matrix migration tests (ovirt/vsphere/openstack)  for every PR in the CI repo.

### CODE REVIEW

**Comment by ahadas** [comment]
@eslutsky can you please elaborate on the changes in the description of the PR?

**Comment by ahadas** [comment]
> adding e2e forklift matrix migration tests (ovirt/vsphere/openstack) for every PR in the CI repo.

but for what purpose? to identify issues?

**Comment by liranr23** [comment]
> > adding e2e forklift matrix migration tests (ovirt/vsphere/openstack) for every PR in the CI repo.
> 
> @eslutsky but for what purpose? to identify issues?

exactly. changes here are affecting the `forklift` CI. better to find the problem in PRs here before merging, letting us know that the PRs in `forklift` solely about the forklift code and not about the framework of the CI.

**Comment by ahadas** [comment]
@liranr23 so if the tests in kubev2v fail, how would you identify here that failures are unrelated to changes in this repo?

**Comment by liranr23** [comment]
> @liranr23 so if the tests in kubev2v/forklift fail, how would you identify here that failures are unrelated to changes in this repo?

e.g the bump to nfs-csi 4.2.0. here we had no troubles but on forklift it fail - you can't really tell if it's because something was changed in the forkliftci or forklift. changes are done first here and then affecting forklift. so if you are doing something here and fail we know forklift will fail as well. while if you fail in forklift you can't. the only problem is if forklift already failing (with merged PR) and then checking here is useless.

File: cluster/providers/utils/deploy_csi_driver_nfs.sh [hunk 1]
```diff
@@ -20,7 +20,7 @@ provisioner: nfs.csi.k8s.io
 parameters:
   server: ${nfs_server_ip}
   share: ${nfs_share}
-  subDir: nfs-csi/\${pvc.metadata.namespace}/\${pvc.metadata.name}  
+  subDir: nfs-csi
```
Comment (liranr23) [inline cluster/providers/utils/deploy_csi_driver_nfs.sh:23]: please rebase, i left it with the subdirs.

**Comment by ahadas** [comment]
@eslutsky the idea of running the e2e tests here feels wrong - maybe to run them post-merge would be ok but I wouldn't duplicate code from kubev2v/forklift that we'll need to keep in-sync for this. I think we should just run the actions in kubev2v/forklift after making a risky change here, and contributors to kubev2v/forklift should be aware of the possibility that changes here could break the actions there

**Comment by eslutsky** [comment]
> @eslutsky the idea of running the e2e tests here feels wrong - maybe running them post-merge would be better but I wouldn't duplicate code from kubev2v/forklift that we'll need to keep in-sync for this. I think we should just run the actions in kubev2v/forklift after making a risky change here, and contributors to kubev2v/forklift should be aware of the possibility that changes here could break the actions there. to me going with this PR is like arguing that when RHEL changes something, they should run tests of RHV, OpenStack and so on...

yes, but you can always go back to previous version of RHEL :)
in case something here breaks the tests in forklift, many developers will be stuck :) ,
i agree it doesn't have to be per PR, and we shouldn't duplicate the code.

**Comment by ahadas** [comment]
> yes, but you can always go back to previous version of RHEL :)

that's exactly what I had in mind this morning - why do we take the latest version of this action rather than specifying a specific commit / tag? this would require bumping up that version and so we could easily tell whether there's been a breaking change here and it will allow us to switch back to a previous state

**Comment by eslutsky** [comment]
> 

+1

**Comment by eslutsky** [comment]
closing this PR  , we will use tags for CI, and bump on forklift action to avoid breaking changes.

**Pull request closed without merging**